### ISSUE TITLE
Toolbar-flex: searchfield with categories

### ISSUE THREAD

**Issue by whernebrink** [infor-design/enterprise#5619]
**Is your feature request related to a problem or use case? Please describe.**
With the tooling we are building (for users to configure their own M3 applications), the users should be able to create an application with a global search, in the application header. The flex-toolbar already support a searchfield inside it, but we also need to be able to have it with categories. 

**Describe the solution you'd like**
We'd like the ability to add a `toolbar-flex-section` e.g.
```html
<soho-toolbar-flex-section class="flex flex-1">
      <label for="searchfield" class="audible">Search</label>
      <input
        soho-searchfield
        id="searchfield"
        name="searchfield"
        class="searchfield"
        [options]="searchfieldOptions"
        [(ngModel)]="model.searchValue"
        (selected)="onCategorySelected()"
      />
 </soho-toolbar-flex-section>
```
 and have it behave and look nice. It would be great if instead of manually adding classes (as in my case "flex flex-1") to say that this flex-sectin should take all possible space, and center its content. Although not required since we can deal wish css in worst case.

**Describe alternatives you've considered**
Here is a [Stackblitz](https://stackblitz.com/edit/angular-ivy-6xlna7?file=src%2Fapp%2Fapp.component.html) (make sure to drag the running example to give it some width) with how it works and look when adding the above markup (it works, but the looks arent great). 

For this example I need to add some flex classes in order for the buttonset to go to the end, and for my searchfield section to be somewhat centered:
![image](https://user-images.githubusercontent.com/73601152/133606661-0e0e76ae-8fef-4111-90ac-b789ec50136b.png)

**Additional context**
This is high up on the wishlist from us at M3 (App designer team). This is what our designer is aiming for:
<img width="1295" alt="image" src="https://user-images.githubusercontent.com/73601152/133606998-0bde14ad-5061-41c8-807f-b4312066c3cc.png">

**Comment by tmcconechy** [comment]
I think for the design it shouldn't be centered as it could cause issues on mobile. Our designers previously came up with this pattern. https://main-enterprise.demo.design.infor.com/components/header/example-searchfield-full

So this would be above or below the other header. with text and buttons.
Perhaps that can be done now but we don't have this example angular but it might be doable. Will that work?

**Comment by whernebrink** [comment]
Are you saying it would then be two toolbars, stacked? One with the search an one with the other buttons and title etc? 

This is our goal to have:
![design](https://user-images.githubusercontent.com/73601152/133622546-0559abdf-a964-41e3-8c03-ca4982b0dd21.png)

How the responsive behaviour is exactly is not important, in our case.

**Comment by tmcconechy** [comment]
Yes im saying that you could stack the example i showed under it so it does take up a bit more vertical space. This is for the case where search is very prominent.
The idea you have with it centered is possible but i feel like then we have too many variations that are similar.

**Comment by tmcconechy** [comment]
OK, will discuss.

**Comment by whernebrink** [comment]
Any feedback on this @tmcconechy?

**Comment by tmcconechy** [comment]
Not right yet I had meant to discuss in the previous planning session however we had a lot of items. Putting this in front of the list for next time and will bring up some discussion sooner

**Comment by tmcconechy** [comment]
@ericangeles adding this one as a priority. I think we just need to make an example out of https://stackblitz.com/edit/angular-ivy-6xlna7?file=src%2Fapp%2Fapp.component.html and find some css fixes to make it

a) work ok with categories
b) fill the center area better

I would probably make an example with categories and without for testing like the stackblitz "design"

**Comment by tmcconechy** [comment]
@kevinwhitedesign does design have any additional feedback on how this should look? if not we can wing it based on current basics. in particular wondering about the different breakpoints in terms of what shows and what doesn't?

I guess just make it fit and if needed truncate the text?

**Comment by ericangeles** [comment]
@tmcconechy do we have a concrete/final design for this one? Or even a prototype design to match expectations?

**Comment by tmcconechy** [comment]
@ericangeles i would go with the suggestions of the M3 team for this case. I.E. https://github.com/infor-design/enterprise/issues/5619#issuecomment-920913059

So that shouldnt change any design we have, its just sizing and responsive and placement in the header

**Comment by whernebrink** [comment]
@ericangeles feel free to tag me and I can assist in testing, if you want. :)

**Comment by janahintal** [comment]
QA Passed. Working as expected in 
http://localhost:4000/components/toolbar-flex/example-toolbar-flex-with-searchfield-in-center.html
http://localhost:4000/components/header/example-toolbar-flex-with-basic-searchfield
http://localhost:4000/components/header/example-toolbar-flex-with-categories.html

**Comment by CindyMercadoReyes** [comment]
BDD attached
[BDD_github#5619.docx](https://github.com/infor-design/enterprise/files/7780806/BDD_github.5619.docx)
.
